src/provider/states/download_package.rs

```diff
@@ -0,0 +1,137 @@
+                // was used to check whether it provides the package
+                info!(
+                    "Starting download of package {} from repository {}",
+                    &package, &repo
+                );
+                let download_directory = pod_state.download_directory.clone();
+                let download_result = repo
+                    .download_package(&package, download_directory.clone())
+                    .await;
+                match download_result {
+                    Ok(()) => {
+                        info!(
+                            "Successfully downloaded package {} to {:?}",
+                            package,
+                            download_directory.clone()
+                        );
+                        return Transition::next(
+                            self,
+                            Installing {
+                                download_directory: pod_state.download_directory.clone(),
+                                parcel_directory: pod_state.parcel_directory.clone(),
+                                package: package.clone(),
+                            },
+                        );
+                    }
+                    Err(e) => {
+                        warn!("Download of package {} failed: {}", package, e);
+                        return Transition::next(
+                            self,
+                            DownloadingBackoff {
+                                package: package.clone(),
+                            },
+                        );
+                    }
+                }
+            }
+            Ok(None) => {
+                // No repository was found that provides this package
+                let message = format!(
+                    "Cannot find package {} in any repository, aborting ..",
+                    &package
+                );
+                error!("{}", &message);
+                return Transition::next(
+                    self,
+                    DownloadingBackoff {
+                        package: package.clone(),
+                    },
+                );
+            }
+            Err(e) => {
+                // An error occurred when looking for a repository providing this package
+                let message = format!(
+                    "Error occurred trying to find package {}: {:?}",
+                    &package, e
+                );
+                error!("{}", &message);
+                return Transition::next(
+                    self,
+                    DownloadingBackoff {
+                        package: package.clone(),
+                    },
+                );
+            }
+        }
+    }
+
+    async fn json_status(
+        &self,
+        _pod_state: &mut PodState,
+        _pod: &Pod,
+    ) -> anyhow::Result<serde_json::Value> {
+        make_status(Phase::Pending, &"status:initializing")
```

Comment:
Do we need the "status:" there? And can we not just use your real status? i.e. "Downloading"

Reply:
I've used the actual status for now, we should have a discussion around how we want to report status at some point in time.